backend/tests/test_course_routes.py

```diff
@@ -0,0 +1,160 @@
+import uuid
+
+from fastapi.testclient import TestClient
+
+from backend.main import app
+from backend.routers.req.courses_req import UpdateCourseReq, CreateCourseReq
+from backend.routers.routes.courses import course_router_endpoint
+
+client = TestClient(app)
+
+email = "test"
+course_id = 1
+test_user_uuid = "e2e422ed-e0b0-4d40-9c92-7d743b11fa83"
+
+
+def test_insert_course():
+    headers = {"X-User-ID": test_user_uuid}
+    test_course = {
+        "id": course_id,
+        "c_group": "test_group",
+        "code": "test_code",
+        "section": "test_term",
+    }
+    response = client.post(course_router_endpoint, json=test_course, headers=headers)
+    assert response.status_code == 200
+
+
+def test_insert_duplicate_course():
+    test_course = {
+        "id": course_id,
+        "c_group": "test_group",
+        "code": "test_code",
+        "section": "test_term",
+    }
+    response = client.post(course_router_endpoint, json=test_course)
+    assert response.status_code == 404
+    assert response.json() == {"detail": "Failed to create course."}
+
+
+# def test_insert_course_invalid_creds():
```

Comment:
Any idea why some of the tests are commented? We could remove them if they're not ready yet.

Reply:
Yes they are not ready, but may need to be tested later to test invalid credentials. I have removed them inshallah.